cfgov/unprocessed/js/routes/on-demand/simple-chart/simple-chart.js

```diff
@@ -206,6 +209,105 @@ function makeFormatter( yAxisLabel ) {
   };
 }
 
+
+/**
+ * Extracts all dates from an object/csv formatted for tilemap display
+ * @param {object} data The data object
+ * @returns {array} Extracted dates
+ * */
+function getTilemapDates( data ) {
+  return Object.keys( data[0] )
+    .filter( k => !isNaN( new Date( k ) ) )
+    .sort( ( a, b ) => new Date( b ) - new Date( a ) );
+}
+
+/**
+ * Builds the tilemap filter DOM
+ * @param {object} chartNode The node where the chart loives
+ * @param {object} chart The chart object
+ * @param {object} data The data object
+ * @param {object} transform Whether data has been transformed
+ */
+function makeTilemapSelect( chartNode, chart, data, transform ) {
+
+  let d;
+  if ( transform ) d = data.transformed;
+  else d = data.raw;
+
+  const options = getTilemapDates( d );
+  const selectNode = makeFilterDOM( options, chartNode, { key: 'tilemap' },
+    'Select a date'
+  );
+
+  attachTilemapFilter( selectNode, chart, data );
+}
+
+
+/**
+ * Wires up the tilemap filter
+ * @param {object} node The created select node
+ * @param {object} chart The chart object
+ * @param {object} data The data object
+ */
+function attachTilemapFilter( node, chart, data ) {
+  node.addEventListener( 'change', evt => {
+    const formatted = formatSeries( data );
+    const updated = getMapConfig( formatted, evt.target.value );
+    chart.update( updated );
+  } );
+}
+
+
+/**
+ * Adds generates a config object to be added to the chart config
+ * @param {array} series The formatted series data
+ * @param {string} date The date to use
+ * @returns {array} series data with a geographic component added
+ * */
+function getMapConfig( series, date ) {
+  let min = Infinity;
+  let max = -Infinity;
+  const data = series[0].data;
+  if ( !date ) date = getTilemapDates( data )[0];
+  const added = data.map( v => {
+    const val = Math.round( Number( v[date] ) * 100 ) / 100;
+    if ( val <= min ) min = val;
+    if ( val >= max ) max = val;
+    return {
+      ...usLayout[v.state_ab],
```

Comment:
I'm curious how [chart-builder](https://github.com/cfpb/cfpb-chart-builder) does the tile layout. Since it sounds like we want our chart to have the same layout as those, is there a way to share the layout code? (Or are we hoping to replace chart-builder with this code eventually?)

Reply:
Chart builder [returns raw svg paths](https://github.com/cfpb/cfpb-chart-builder/blob/main/src/js/utils/get-tile-map-state.js), which I kinda feel is a step down from a grid-based approach which on the whole is more easily configurable.

Reply:
Makes sense to me 👍